models/user.go

```diff
@@ -86,37 +74,70 @@ func generateVerificationCode() string {
 	return string(rst)
 }
 
+// IsLastVerificationCodeGeneratedTimeExceeded checks if the current is within the duration allowed
+// starting from when the latest verification code is created
+func (t TX) IsLastVerificationCodeGeneratedTimeExceeded(u string) (bool, error) {
+	var result EmailVerification
+	err := t.tx.Where(&EmailVerification{Email: u}).Order("expired_at desc").First(&result).Error
```

Comment:
Just another thought, don't need to change the current one.

1.  maybe add a `CreatedAt` field instead of using `ExpiredAt` (or even drop `ExpiredAt`).
2. do something like `t.tx.Where(&EmailVerification{Email: u}).Where("expired_at < ?", threasholdTime).Count()` ?